### PR TITLE
Implement `replace_segment` attribute

### PR DESCRIPTION
**Pull request by bkchr**
This implements the `replace_segment` attribute. This attribute can be used to rename one or multiple segments returned by `TypeInfo::path`. It could be for example be used to rename a changed crate name. For more info see the examples and the docs in this pr.

### CODE REVIEW

**Review by franciscoaguirre — APPROVED** [review]

**Review by KiChjang — APPROVED** [review]

File: derive/src/lib.rs [hunk 1]
```diff
@@ -101,12 +101,23 @@ impl TypeInfoImpl {
...
+        });
+
         Ok(quote! {
             impl #impl_generics #scale_info::TypeInfo for #ident #ty_generics #where_clause {
                 type Identity = Self;
                 fn type_info() -> #scale_info::Type {
                     #scale_info::Type::builder()
-                        .path(#scale_info::Path::new(::core::stringify!(#ident), ::core::module_path!()))
+                        .path(#scale_info::Path::new_with_replace(
+                            ::core::stringify!(#ident),
```
Comment (ggwpez) [inline derive/src/lib.rs:117]: Why not pass the ident as `syn::Ident`? I think it would be cleaner than calling `.split("::")` subsequently.

**Review by ggwpez — APPROVED** [review]
Just one nit.

File: derive/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply derive/src/lib.rs:117]: We call `split` on the module path that comes out of the `module_path!` macro. Or I maybe misunderstand you?

**Review by acatangiu — APPROVED** [review]

**Review by lexnv — APPROVED** [review]
LGTM! 

Small nit: we'd also need an entry in the [changelog](https://github.com/paritytech/scale-info/blob/master/CHANGELOG.md) if we plan to release from this PR 🙏

File: derive/src/lib.rs (hunk 1 above)
Comment (ggwpez) [reply derive/src/lib.rs]: Oh, yea :man_facepalming:

**Pull request merged**